src/Component/src/Metadata/Mutator/ResourceMutatorCollection.php

```
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Metadata\Mutator;
+
+use Sylius\Resource\Metadata\ResourceMutatorInterface;
+
+/**
+ * @internal
+ */
+final class ResourceResourceMutatorCollection implements ResourceMutatorCollectionInterface
+{
+    private array $mutators;
+
+    public function add(string $resourceClass, ResourceMutatorInterface $mutator): void
+    {
+        $this->mutators[$resourceClass][] = $mutator;
+    }
+
+    public function get(string $id): array
```

When resource mutator is not found, I would probably expect `null` value, not empty array, but since it's internal and probably used in some `foreach`, then it's okay as is.

---

yep, it's used on a `foreach`, it's marked as internal.
Maybe should we throw an exception later.